### PR TITLE
Update Docusaurus to v3

### LINKED ISSUE #3748: Update Docusaurus to v3
### What should be done?

We should migrate from Docusaurus v2 to v3.
https://docusaurus.io/blog/releases/3.0

### Where?

Deps, docs, blog.

### Definition of Done

- Docusaurus v3 is used;
- nothing is broken.

### PR DESCRIPTION
**Pull request by Fashander**
# Description

Closes #3748.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: build/deps/docusaurus-docs.Dockerfile [hunk 1]
```diff
@@ -1 +1 @@
-FROM ghcr.io/ferretdb/docusaurus-docs:2.4.3-3
+FROM ghcr.io/ferretdb/docusaurus-docs:3.0.0-1
```
Comment (AlekSi) [inline build/deps/docusaurus-docs.Dockerfile:1]: Please add a line break there

File: website/blog/2023-01-25-ferretdb-fetches-data-query-pushdown.md [hunk 2]
```diff
@@ -98,7 +98,7 @@ This is needed for our internal representation of the documents called `pjson`,
 With these documents, FerretDB can parse the data to internal `pjson` type, and iterate through all of them to apply the filters.
 After this process, as only one document matches the filter, the projection will be applied to it, so only the `active` (and `_id`) fields will be returned:
 
-```json5
+```json
```
Comment (AlekSi) [inline website/blog/2023-01-25-ferretdb-fetches-data-query-pushdown.md:101]: We used `json5` because that's not a valid JSON: unquoted field names can't rendered correctly by `json`. We should enable `json5`, not change it to `json`.

File: website/docs/configuration/flags.md [hunk 3]
```diff
@@ -58,9 +58,9 @@ Some default values are overridden in [our Docker image](../quickstart-guide/doc
 FerretDB uses [pgx v5](https://github.com/jackc/pgx) library for connecting to PostgreSQL.
 Supported URL parameters are documented there:
 
-- <https://pkg.go.dev/github.com/jackc/pgx/v5/pgconn#ParseConfig>
-- <https://pkg.go.dev/github.com/jackc/pgx/v5#ParseConfig>
-- <https://pkg.go.dev/github.com/jackc/pgx/v5/pgxpool#ParseConfig>
+- [https://pkg.go.dev/github.com/jackc/pgx/v5/pgconn#ParseConfig](https://pkg.go.dev/github.com/jackc/pgx/v5/pgconn#ParseConfig)
```
[the remark this answers is not in the document]
Comment (Fashander) [reply website/docs/configuration/flags.md:61]: https://docusaurus.io/blog/preparing-your-site-for-docusaurus-v3#bad-usage-of--1

**Comment by Fashander** [comment]
`json5` extends `json`, so we need to load it before it can actually work, otherwise, it returns an error of `undefined`

File: website/blog/2022-11-08-pjson-how-to-store-bson-in-jsonb.md [hunk 4]
```diff
@@ -32,8 +32,8 @@ Here the hexadecimal `\x00` notation represents `0000 0000` in bits and similarl
 In BSON, `\x00` is a byte used as a terminator to indicate the end of a document.
 In BSON, field names use `cstring` which are UTF-8 characters followed by `\x00`.
 
-```json
-{ "foo": "bar" }
+```json5
+{ foo: 'bar' }
```
Comment (AlekSi) [inline website/blog/2022-11-08-pjson-how-to-store-bson-in-jsonb.md:36]: I think we should not make unrelated changes in this PR. Both forms should work.

File: website/docs/configuration/flags.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (AlekSi) [reply website/docs/configuration/flags.md:64]: I think we should disable that rule in the linter: https://github.com/updownpress/markdown-lint/blob/master/rules/034-no-bare-urls.md
